tools/dtrace/slowest-queries.d

```diff
@@ -0,0 +1,41 @@
+#!/usr/sbin/dtrace -qs
+
+#pragma strsize=4k
+
+/* Print out the slowest 5 queries every 10 seconds. */
+
+dtrace:::BEGIN
+{
+    printf("Tracing slowest queries for nexus PID %d, use Ctrl-C to exit\n", $target);
+}
+
+diesel-db$target:::query-start
+{
+    this->conn_id = json(copyinstr(arg1), "ok");
+    ts[this->conn_id] = timestamp;
+    query[this->conn_id] = copyinstr(arg2);
+}
+
+
+diesel-db$target:::query-done
+{
+    this->conn_id = json(copyinstr(arg1), "ok");
+}
+
+diesel-db$target:::query-done
+/ts[this->conn_id]/
+{
+    this->latency = timestamp - ts[this->conn_id];
+    @[this->conn_id, query[this->conn_id]] = max(this->latency);
```

Comment:
This use of max is cute enough that it might warrant a comment to the effect that since conn_id is unique max will never compare etc